lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -304,10 +304,10 @@ def test_partial_period(self, warning_list=None):
             " accumulation_period, i.e. the number of cubes is insufficient to"
             " give a set of complete periods. Only complete periods will be"
             " returned.")
-        upper_bound_fp = self.cubes[5].coord("forecast_period").points
-        integral = 0
+        upper_bound_fp = self.cubes[8].coord("forecast_period").points
+        integral = 10
         plugin = Accumulation(
-            accumulation_period=5*60,
+            accumulation_period=10*60,
```

Comment:
This is better. The ```upper_bound_fp = self.cubes[8]``` didn't need to change. Effectively we want a test that is as close to the one above as possible, it simply needs a longer accumulation period (integral = 10, accumulation_period = 10*60) such that there are not enough cubes to complete a 10 minute period ending at ```self.cubes[5].coord('forecast_period')```.

But wait!

This highlights another issue. This warning is now going to be raised whenever accumulation_period > lead_time_interval, which could be every cycle if the suite is set up to produce 15 minute rates and 1 hour accumulations. This was not originally possible as these two values (accumulation period and lead time interval) were passed as the same value from the CLI.

Do we need this warning in its current form? I think we should instead rewrite the warning such that it is only raised if no accumulation period will be returned at all (i.e. if integral is 0 when it is calculated).

Reply:
I've reworked this checking because as you point out, this warning would otherwise always be raised, if we are generating 1 hour accumulations at forecast periods incrementing every 15 minutes. I've also added a filtering, so that we only try to compute accumulations at forecast periods greater than the accumulation period.